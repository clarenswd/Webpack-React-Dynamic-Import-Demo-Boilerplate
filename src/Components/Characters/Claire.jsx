import React from 'react';
import Loadable from 'react-loadable';
import LoadingMessage from '../Common/LoadingMessage';
import Knife from '../Weapons/Knife';
// import Magnum from '../Weapons/Magnum';

const Magnum = Loadable({
  loader: () => import('../Weapons/Magnum' /* webpackChunkName: 'Magnum' */),
  loading: LoadingMessage,
});

const Claire = () => {
  const name = 'Claire Redfield';

  return (
    <div>
      <h2>{name}</h2>
      <ul>
        <li><Knife /></li>
        <li><Magnum /></li>
      </ul>
    </div>
  );
};

export default Claire;
