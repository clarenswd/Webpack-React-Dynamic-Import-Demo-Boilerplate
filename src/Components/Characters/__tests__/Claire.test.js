import React from 'react';
import renderer from 'react-test-renderer';
import Claire from '../Claire';

it('renders correctly', () => {
  const tree = renderer
    .create(<Claire />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
