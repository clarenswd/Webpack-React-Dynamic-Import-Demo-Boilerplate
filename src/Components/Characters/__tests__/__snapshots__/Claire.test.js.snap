// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2>
    Claire Redfield
  </h2>
  <div
    className="container"
  >
    <div
      className="knife"
    >
      Knife
    </div>
    <div
      className="loadingMessage"
    >
      Loading...
    </div>
  </div>
</div>
`;
